src/util/iroha-util.js

```diff
@@ -357,31 +357,43 @@ function command (
     })
   })
     .then(() => {
-      debug('sleep 5 seconds...')
-      return sleep(5000)
-    })
-    .then(() => {
-      debug('sending transaction status request...')
+      debug('opening transaction status stream...')
 
       return new Promise((resolve, reject) => {
         const request = new TxStatusRequest()
 
         request.setTxHash(txHash)
 
-        txClient.status(request, (err, response) => {
-          if (err) {
-            return reject(err)
-          }
+        let stream = txClient.statusStream(request)
+
+        // TODO: fix this if (when) https://github.com/improbable-eng/grpc-web/pull/189 is merged
+        const notaryError = () => {
+          reject(new Error('Problem with notary. Please try again later.'))
+          stream.cancel()
+        }
+        let timer = setTimeout(notaryError, timeoutLimit * 2)
+
+        let statuses = []
 
-          const status = response.getTxStatus()
-          const statusName = getProtoEnumName(
+        stream.on('data', function (response) {
+          clearInterval(timer)
```

Comment:
I think it should be clearTimeout - but I found that a timer can be stopped with either clearTimeout or clearInterval. 😲

Reply:
You are right, I should rather change this since it can lead to some problems https://stackoverflow.com/questions/9913719/are-cleartimeout-and-clearinterval-the-same